routes/course.js

```diff
@@ -223,37 +246,53 @@ router.patch("/api/courses/changeDistribution/:course_id"),
   };
 */
 
-//delete a course given course id
-//update associated distribution credits
+// delete a course given course id
+// update associated distribution credits
 router.delete("/api/courses/:course_id", (req, res) => {
-  const c_id = req.params.course_id;
+  const c_id = ObjectId(req.params.course_id);
   courses
     .findByIdAndDelete(c_id)
-    .then((course) => {
-      course.distribution_ids.forEach((id) => {
-        distributions
-          .findByIdAndUpdate(
-            id,
-            { $pull: { courses: c_id } },
-            { new: true, runValidators: true }
-          )
-          .then((distribution) =>
-            distributionCreditUpdate(distribution, course, false)
-          )
-          .catch((err) => errorHandler(res, 500, err));
-      });
+    .then(async (course) => {
+      let updatedDists = [];
+      let updatedFines = [];
+      // remove course from distributions
+      for (let id of course.distribution_ids) {
+        await distributions
+          .findById(id)
+          .then(async (distribution) => {
+            await requirementCreditUpdate(distribution, course, false);
+            // remove course from fineReqs
+            for (let f_id of distribution.fineReq_ids) {
+              let fine = await fineRequirements.findById(f_id);
+              if (checkCriteriaSatisfied(fine.criteria, course)) {
+                await requirementCreditUpdate(fine, course, false);
+                updatedFines.push(fine);
+              }
+            }
+            // determine distribution satisfied with pathing
+            if (distribution.planned >= distribution.required_credits) {
+              if (distribution.pathing) {
+                await processPathing(distribution);
+              } else {
+                distribution.satisfied = true;
+              }
+            }
+            await distribution.save();
+            updatedDists.push(distribution);
+          })
+      }
       //delete course id to user's year array
       let query = {};
       query[course.year] = course._id; //e.g. { freshman: id }
```

Comment:
please consider remove the course_ids field in the year schema

Reply:
Good catch!